### ISSUE TITLE
SCREEN_PIXEL_SIZE not working properly

### ISSUE THREAD

**Issue by 11clock** [godotengine/godot#16312]
**Godot version:**
Godot 3.0 Stable

**OS/device including version:**
Windows 10

**Issue description:**
When I try to use SCREEN_PIXEL_SIZE as opposed to hardcoding the resolution, my shader no longer works properly.

**Steps to reproduce:**
Add this vignette shader code to a TextureRect that covers the entire screen with a pure black texture and scaled. Replace the first resolution's initialization with your project's resolution. Run the scene that the shader is in. Afterwards, comment the first resolution initialization and uncomment the second one. Then play the scene again. Notice how the shader just flat out makes the screen darker, when behavior should be near identical.
```
shader_type canvas_item;

void fragment() {
        float radius = 0.75;
        float softness = 0.45;
        float intensity = 0.15;
	vec2 resolution = vec2(640.0, 360.0);
	//vec2 resolution = SCREEN_PIXEL_SIZE;
	
	vec4 pixel_color = texture(SCREEN_TEXTURE, SCREEN_UV);
	vec2 position = (FRAGCOORD.xy / resolution.xy) - vec2(0.5);
	float len = length(position);
	float vignette = smoothstep(radius, softness, len);
	pixel_color.rgb = mix(pixel_color.rgb, pixel_color.rgb * vignette, intensity);
	COLOR = pixel_color;
}
```

**Comment by binbitten** [comment]
_SCREEN_PIXEL_SIZE_ is the inverted resolution, i.e. vec2(1.0 / width, 1.0 / height).

**Comment by akien-mga** [comment]
It would be nice to improve its description in the docs then: http://docs.godotengine.org/en/latest/tutorials/shading/shading_language.html?highlight=SCREEN_PIXEL_SIZE#id4

**Comment by clayjohn** [comment]
This issue should be closed now.

**Comment by akien-mga** [comment]
Fixed by godotengine/godot-docs#1443.
